src/foam/u2/wizard/StepWizardController.js

```diff
@@ -0,0 +1,321 @@
+          });
+          var isAvailable$ = section.isAvailable$;
+          this.wsub.onDetach(isAvailable$.sub(() => {
+            console.log('Y');
+            this.onWizardPositionAvailabilityUpdate(
+              sectionPosition, isAvailable$.get());
+          }));
+        });
+
+      });
+
+    },
+    function detach() {
+      this.wsub.detach();
+      this.SUPER();
+    },
+    function positionAfter(pos) {
+      let subWi = pos.wizardletIndex
+      let subSi = pos.sectionIndex;
+      if ( subSi >= this.wizardlets[subWi].sections.length - 1 ) {
+        if ( subWi >= this.wizardlets.length - 1 ) return null;
+        subSi = 0;
+        subWi++;
+      } else {
+        subSi++;
+      }
+      return this.WizardPosition.create({
+        wizardletIndex: subWi,
+        sectionIndex: subSi,
+      });
+    },
+    function positionBefore(pos) {
+      let subWi = pos.wizardletIndex;
+      let subSi = pos.sectionIndex;
+      if ( subSi == 0 ) {
+        if ( subWi == 0 ) return null;
+        subWi--;
+        // Skip past steps with no sections
+        while ( this.wizardlets[subWi].sections.length < 1 ) subWi--;
+        if ( subWi < 0 ) return null;
+        subSi = this.wizardlets[subWi].sections.length - 1;
+      } else {
+        subSi--;
+      }
+      return this.WizardPosition.create({
+        wizardletIndex: subWi,
+        sectionIndex: subSi,
+      });
+    },
+    function nextAvailable(pos, iter) {
+      for ( let p = iter(pos) ; p != null ; p = iter(p) ) {
+        let wizardlet = this.wizardlets[p.wizardletIndex]
+        if ( ! wizardlet.isVisible ) continue;
+
+        if (
+          wizardlet.sections.length > 0 &&
+          wizardlet.sections[p.sectionIndex].isAvailable
+        ) return p;
+      }
+
+      return null;
+    },
+    function saveProgress() {
+      return this.wizardlets.reduce((p, wizardlet) =>
+        this.visitedWizardlets.indexOf(wizardlet) == -1 ? p
+          : p.then(() => wizardlet.save()), Promise.resolve());
+    },
+    function next() {
+      // Save current wizardlet, and any save-able (isAvailable) but invisible
+      // wizardlets that may exist in between this one and the next.
+      var start = this.wizardPosition.wizardletIndex;
+      var end = this.nextScreen ?
+        this.nextScreen.wizardletIndex : this.wizardlets.length;
+      var p = Promise.resolve();
+      for ( let i = start ; i < end ; i++ ) {
+        if ( ! this.wizardlets[i].isAvailable ) continue;
+        p = p.then(() => this.wizardlets[i].save());
+      }
+      return p.then(() => {
+        if ( this.nextScreen == null ) {
+          this.submitted = true;
+          return true;
+        }
+        this.wizardPosition = this.nextScreen;
+        return false;
+      });
+    },
+    function back() {
+      let previousScreen = this.previousScreen;
+      if ( previousScreen == null ) {
+        throw new Error('back() called without checking canGoBack');
```

Comment:
Where does this get called from? If this is a message a user can see this needs to be a message and need to add local support.

Reply:
This error case is a developer error so an exception makes sense here